ucs/clientlist.cpp

```diff
@@ -594,6 +596,35 @@ void Clientlist::CheckForStaleConnections(Client *c) {
 	}
 }
 
+std::string RemoveDuplicateChannels(std::string InChannels) {
+	// Split the string by ", " and store the names in a vector
+	std::vector<std::string> channel_names;
+	size_t start = 0, end = 0;
+	while ((end = InChannels.find(", ", start)) != std::string::npos) {
+		channel_names.push_back(InChannels.substr(start, end - start));
+		start = end + 2;
+	}
+	channel_names.push_back(InChannels.substr(start)); // Add the last channel
+
+	// Remove duplicates by inserting the names of the channels into an unordered set
+	// and then copying the unique elements back into the original vector
+	std::unordered_set<std::string> unique_channels;
+	channel_names.erase(std::remove_if(channel_names.begin(), channel_names.end(),
+		[&unique_channels](const std::string& channel) {
+			return !unique_channels.insert(channel).second;
+		}), channel_names.end());
+
+	// Concatenate the names of the unique channels into a single string
```

Comment:
We have `Strings::Implode` for this exact functionality, will keep you from having to chop the comma off the end as well

Reply:
Fixed